iroh-net/src/key.rs

```diff
@@ -116,8 +116,8 @@ impl<'de> Deserialize<'de> for PublicKey {
         D: serde::Deserializer<'de>,
     {
         if deserializer.is_human_readable() {
-            let s: &str = serde::Deserialize::deserialize(deserializer)?;
-            Self::from_str(s).map_err(serde::de::Error::custom)
+            let s = String::deserialize(deserializer)?;
```

Comment:
why was this needed?

Reply:
This is unrelated to the discovery. Without this I had issues deserializing from toml using our toml lib. Deserializing as a `&str` is more restrictive than deserializing as a String since it is zero copy, so you can not do it from an owned String, but only from a &str.